### PR TITLE
check if the lump can be a Doom patch in R_GenerateLookup

### LINKED ISSUE #616: segfault when loading Urotsuki.wad
```gdb
Program received signal SIGSEGV, Segmentation fault.
R_GenerateLookup (errors=<synthetic pointer>, texnum=563)
    at /home/fabian/Source/woof/src/r_data.c:408
408			for (;col->topdelta != 0xff; count[x].posts++)
(gdb) l
403			const column_t *col =
404			  (column_t*)((byte*) realpatch+LONG(cofs[x]));
405			const byte *base = (const byte *) col;
406	
407			// count posts
408			for (;col->topdelta != 0xff; count[x].posts++)
409			  if ((unsigned)((byte *) col - base) <= limit)
410			    col = (column_t *)((byte *) col + col->length + 4);
411			  else
412			    { // killough 12/98: warn about column construction bug
```

Culprit is texture number 563:
```
(gdb) p *textures[563]
$8 = {
  name = "TEKBRFLT",
  next = -640034343,
  index = -640034343,
  width = 64,
  height = 64,
  patchcount = 1,
  patches = {{
      originx = 0,
      originy = 0,
      patch = 1701
    }}
}
```

### PR DESCRIPTION
**Pull request by rfomin**
This change fixed crash on load with Urotsuki.wad [[idgames]](https://www.doomworld.com/idgames/levels/doom2/Ports/s-u/urotsuki)
Reported in [[doomworld]](https://www.doomworld.com/forum/topic/112333-this-is-woof-1000-jun-17-2022-updated-winmbf/?do=findComment&comment=2509503)

Fixes #616

### CODE REVIEW

**Comment by fabiangreffrath** [comment]
Hm, but it works for any other texture from any other WAD. What's wrong with this one. I don't want to give up that code.

**Comment by rfomin** [comment]
> Hm, but it works for any other texture from any other WAD. What's wrong with this one. I don't want to give up that code.

I still don't know how this code works, but it crash here.

**Comment by fabiangreffrath** [comment]
Yes, it's complicated. Does it also crash here in Crispy?

**Comment by rfomin** [comment]
> Yes, it's complicated. Does it also crash here in Crispy?

It silently doesn't load (in Woof 9.0.0 too)

File: src/r_data.c [hunk 1]
```diff
@@ -391,6 +391,9 @@ static void R_GenerateLookup(int texnum, int *const errors)
 	  int x, x1 = patch++->originx, x2 = x1 + SHORT(realpatch->width);
 	  const int *cofs = realpatch->columnofs - x1;
 	  
+	  if (!R_IsPatchLump(pat))
+	    continue;
```
Comment (fabiangreffrath) [inline src/r_data.c:395]: I'd like to report texture, patch name and number to stdout in this case.

Comment (fabiangreffrath) [reply src/r_data.c:398]: Ah, sorry, but stderr makes more sense here. Also, could you print the function name instead of "Warning"?

**Comment by fabiangreffrath** [comment]
Thank you!

**Pull request merged**